modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/system/info/collection/provider/test/SystemObjectEntrySingleFormVariationInfoCollectionProviderTest.java

```diff
@@ -0,0 +1,172 @@
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.portal.test.rule.FeatureFlags;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+
+import java.util.List;
+import java.util.Map;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.mock.web.MockHttpServletRequest;
+
+/**
+ * @author Carolina Barbosa
+ */
+@FeatureFlags("LPD-17965")
+@RunWith(Arquillian.class)
+public class SystemObjectEntrySingleFormVariationInfoCollectionProviderTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		ServiceContextThreadLocal.pushServiceContext(_getServiceContext());
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		ServiceContextThreadLocal.popServiceContext();
+	}
+
+	@Test
+	public void testGetCollectionInfoPage() throws Exception {
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.fetchObjectDefinition(
+				TestPropsValues.getCompanyId(),
+				Organization.class.getSimpleName());
+
+		SystemObjectDefinitionManager systemObjectDefinitionManager =
+			_systemObjectDefinitionManagerRegistry.
+				getSystemObjectDefinitionManager(objectDefinition.getName());
+
+		long organizationId1 = systemObjectDefinitionManager.addBaseModel(
+			TestPropsValues.getUser(),
+			HashMapBuilder.<String, Object>put(
+				"comment", "Comment 1"
+			).put(
+				"name", "Name 1"
+			).build());
+		long organizationId2 = systemObjectDefinitionManager.addBaseModel(
+			TestPropsValues.getUser(),
+			HashMapBuilder.<String, Object>put(
+				"comment", "Comment 2"
+			).put(
+				"name", "Name 2"
+			).build());
+
+		InfoCollectionProvider<SystemObjectEntry> infoCollectionProvider =
+			_infoItemServiceRegistry.getFirstInfoItemService(
+				InfoCollectionProvider.class,
+				objectDefinition.getClassName() + StringPool.POUND +
+					objectDefinition.getObjectDefinitionId());
+
+		CollectionQuery collectionQuery = new CollectionQuery();
+
+		collectionQuery.setPagination(Pagination.of(1, 0));
+
+		InfoPage<SystemObjectEntry> infoPage =
+			infoCollectionProvider.getCollectionInfoPage(collectionQuery);
+
+		Assert.assertEquals(2, infoPage.getTotalCount());
+
+		_assertSystemObjectEntries(
+			organizationId1, "Comment 1", "Name 1", infoPage);
+
+		collectionQuery.setPagination(Pagination.of(2, 1));
+
+		_assertSystemObjectEntries(
+			organizationId2, "Comment 2", "Name 2",
+			infoCollectionProvider.getCollectionInfoPage(collectionQuery));
+	}
+
+	private void _assertSystemObjectEntries(
```

Comment:
The method name is plural, but you assert one object entry. What do you think about renaming this method?

Reply:
this method validates 1 entry which has 2 values (commet, name)